File: 0525-contiguous-array/0525-contiguous-array.py
class Solution:
    def findMaxLength(self, nums: List[int]) -> int:
        temp = 0
        ans = 0
        hm = {0: -1} 

        for i in range(len(nums)):
            curr = nums[i]
            temp += -1 if curr == 0 else 1
            
            if temp in hm:
                ans = max(ans, i - hm[temp])
            else:
                hm[temp] = i 
            
        return ans

File: 0525-contiguous-array/0525-contiguous-array.cpp
class Solution {
public:
    int findMaxLength(vector<int>& nums) {
        unordered_map<int, int> hm;
        hm[0] = -1;
        int temp = 0;
        int ans = 0;
        
        for (int i = 0; i < nums.size(); i++) {
            temp += (nums[i] == 0) ? -1 : 1;
            
            if (hm.find(temp) != hm.end()) {
                ans = std::max(ans, i - hm[temp]);
            } else {
                hm[temp] = i;
            }
        }
        
        return ans;
    }
};